### PR TITLE
bt-311: [BUG] User is redirected to onboarding when forgot password is clicked

### LINKED ISSUE #311: [BUG] User is redirected to the onboarding steps after clicking on "Forgot Password?"
**Description:**
User is redirected to the onboarding steps after clicking on "Forgot Password?" on the Sign in page

**Environment:**
Windows 11
Google Chrome 116.0

**Steps to reproduce:**
1.  Open the page http://localhost:3000/sign-in
2. Click on "Forgot Password?"

**Actual Result:**
User is redirected to the onboarding steps after clicking on "Forgot Password?" on the Sign in page

**Expected Result:**
User is redirected to the page to recover the password  after clicking on "Forgot Password?"

**Visual Proof (screenshots, videos, text):**
image

**Severity:**
Major (S3)

### PR DESCRIPTION
**Pull request by JosuerBague**
closes #311 

Made a temporary reset password page.

This design is not from official Figma design and serves only as a mock placeholder.

### CODE REVIEW

**Comment by nikita-remeslov** [comment]
@JosuerBague could you provide some screenshots?

**Comment by JosuerBague** [comment]
> @JosuerBague could you provide some screenshots?

image

image

**Comment by nikita-remeslov** [comment]
Discussed in Slack

**Comment by JosuerBague** [comment]
New layout:

Wide Screen:
image

Mobile:
image

File: frontend/src/bundles/auth/components/auth-forms/sign-in-form/sign-in-form.tsx [hunk 1]
```diff
@@ -78,11 +78,11 @@ const SignInForm: React.FC<Properties> = ({ onSubmit }) => {
                         className={styles.checkbox}
                     />
                     {/* TODO: Link to reset password route */}
-                    <Link to="/" className={styles.forgot}>
+                    <Link to="/reset-password" className={styles.forgot}>
```
Comment (nikita-remeslov) [inline frontend/src/bundles/auth/components/auth-forms/sign-in-form/sign-in-form.tsx]: we can use enum here

Comment (anton-otroshchenko) [inline frontend/src/bundles/auth/components/auth-forms/sign-in-form/sign-in-form.tsx]: Use our AppRoute enum

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

**Comment by nikita-remeslov** [comment]
make the same styles for header
image

**Review by savinaDaria — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**